src/main/java/dk/kb/util/other/StringListUtils.java

```diff
@@ -0,0 +1,252 @@
+                false);
+    }
+    
+    /**
+     * Return the string or "" if null
+     *
+     * @param string the string
+     * @return same or "" if null
+     */
+    public static String notNull(String string) {
+        return Optional.ofNullable(string).orElse("");
+    }
+    
+    public static String firstOf(List<String> list, String defaultValue) {
+        if (list.isEmpty()) {
+            return defaultValue;
+        } else {
+            return list.get(0);
+        }
+    }
+    
+    public static String getParamOrProps(Properties props, String param, String propKey) {
+        if (param == null) {
+            
+            if (!props.contains(propKey)) {
+                param = props.getProperty(propKey);
+            } else {
+                param = null;
+            }
+        }
+        return param;
+    }
+    
+    public static List<String> removeEmpties(List<String> list) {
+        if (list == null) {
+            return Collections.emptyList();
+        }
+        return list.stream()
+                   .filter(Objects::nonNull)
+                   .filter(number -> !number.trim().isEmpty())
+                   .distinct()
+                   .collect(Collectors.toList());
+    }
+    
+    public static <T extends Iterable<String>> T removeSubstrings(T list) {
+        
+        Iterator<String> firstIterator = list.iterator();
+        while (firstIterator.hasNext()) {
+            String s = firstIterator.next();
+            for (String t : list) {
+                if (s.equals(t)) {
+                    continue;
+                }
+                if (t.contains(s)) {
+                    firstIterator.remove();
+                    break;
+                }
+            }
+        }
+        
+        return list;
+    }
+    
+    @SafeVarargs
+    public static <T> Set<T> setOf(T... objects) {
+        return new HashSet<T>(Arrays.asList(objects));
+    }
+    
+    //
+    //public static Set<String> removeSubstrings(Set<String> list) {
+    //
+    //    Set<String> result = new HashSet<>();
+    //    for (String s : list) {
+    //        boolean substring = false;
+    //        for (String t : list) {
+    //            if (s.equals(t)){
+    //                continue;
+    //            }
+    //            if (t.contains(s)){
+    //                substring = true;
+    //                break;
+    //            }
+    //        }
+    //        if (!substring){
+    //            result.add(s);
+    //        }
+    //    }
+    //    return result;
+    //}
+    private static List<String> removeEmpties(String... list) {
+        return Arrays.stream(list)
+                     .filter(Objects::nonNull)
+                     .filter(number -> !number.trim().isEmpty())
+                     .distinct()
+                     .collect(Collectors.toList());
+    }
+    
+    public static String orDefault(String value, String default_value) {
```

Comment:
The name does not signal what happens (check for null or empty).

Reply:
Renamed to useDefaultIfNullOrEmpty